src/components/site-runners.tsx

```diff
@@ -27,15 +33,15 @@ export function RunnerProvider({
   const addSite = useCallback(
     (site: GatsbySite) => {
       sites.set(site.root, site)
-      setSites(sites)
+      setSites(new Map(sites))
```

Comment:
It's not quite clear to me right now why this is changed to a shallow clone

Reply:
So that React knows it's changed and re-renders

Reply:
Ah yes, that makes sense.